crmsh/bootstrap.py

```diff
@@ -105,6 +106,225 @@ def init_qdevice(self):
+
+    def _update_configuration(self):
+        """
+        Update /etc/sysconfig/sbd
+        """
+        shutil.copyfile(self.SYSCONFIG_SBD_TEMPLATE, SYSCONFIG_SBD)
+        sbd_config_dict = {
+                "SBD_PACEMAKER": "yes",
+                "SBD_STARTMODE": "always",
+                "SBD_DELAY_START": "no",
+                "SBD_WATCHDOG_DEV": detect_watchdog_device()
+                }
+        if self._sbd_devices:
+            sbd_config_dict["SBD_DEVICE"] = ';'.join(self._sbd_devices)
+        utils.sysconfig_set(SYSCONFIG_SBD, **sbd_config_dict)
+        csync2_update(SYSCONFIG_SBD)
+
+    @staticmethod
+    def _get_sbd_device_from_config():
+        """
+        Gets currently configured SBD device, i.e. what's in /etc/sysconfig/sbd
+        """
+        conf = utils.parse_sysconfig(SYSCONFIG_SBD)
+        res = conf.get("SBD_DEVICE")
+        if res:
+            return res.strip(';').split(';')
+        else:
+            return None
+
+    def sbd_init(self):
+        """
+        Function sbd_init includes these steps:
+        1. Get sbd device from options or interactive mode
+        2. Initialize sbd device
+        3. Write config file /etc/sysconfig/sbd
+        """
+        self._get_sbd_device()
+        if not self._sbd_devices and not self.diskless_sbd:
+            return
+        status_long("Initializing {}SBD...".format("diskless " if self.diskless_sbd else ""))
+        self._initialize_sbd()
+        self._update_configuration()
+        status_done()
+        # If process work through here, consider it's ready for enable service
+        self._sbd_service_flag = True
+
+    def manage_sbd_service(self):
+        """
+        Manage sbd service, running on both init and join process
+        """
+        if self._sbd_service_flag:
+            invoke("systemctl enable sbd.service")
+        else:
+            invoke("systemctl disable sbd.service")
+
+    def configure_sbd_resource(self):
+        """
+        Configure stonith-sbd resource and stonith-enabled property
+        """
+        if self._sbd_devices and self._get_sbd_device_from_config():
+            if not invoke("crm configure primitive stonith-sbd stonith:external/sbd pcmk_delay_max=30s"):
+                error("Can't create stonith-sbd primitive")
+            if not invoke("crm configure property stonith-enabled=true"):
+                error("Can't enable STONITH for SBD")
+        elif self.diskless_sbd:
+            if not invoke("crm configure property stonith-enabled=true stonith-watchdog-timeout=5s"):
+                error("Can't enable STONITH for diskless SBD")
+
+    def join_sbd(self, peer_host):
+        """
+        Function join_sbd running on join process only
+        On joining process, check whether peer node has enabled sbd.service
+        If so, check prerequisites of SBD and verify sbd device on join node
+        """
+        if not os.path.exists(SYSCONFIG_SBD):
```

Comment:
Just wondering,  does it mean /etc/sysconfig/sbd is supposed to have already been synchronized to this joining node if there's any on the first node?